src/react-components/src/__tests__/utils/bem.test.ts

```diff
@@ -0,0 +1,32 @@
+import { expect } from "chai";
+import bem from "../../utils/bem";
+
+describe("given a baseclass", () => {
+    it("returns baseclass", () => {
+        expect(bem("unorder-list")).to.equal("unorder-list");
+    });
+});
+
+describe("given a baseclass with modifier and blockName", () => {
+    it("returns properly formatted baseclass, modifier and blockname", () => {
+        expect(bem("list-item", ["scroll"], "unordered-list")).to.equal(
+            "unordered-list__list-item unordered-list__list-item--scroll"
+        );
+    });
+});
+
+describe("given a baseclass with modifier and no blockName", () => {
+    it("returns properly formatted baseclass and modifier", () => {
+        expect(bem("list-item", ["scroll"])).to.equal(
+            "list-item list-item--scroll"
+        );
+    });
+});
+
+describe("given a baseclass with no modifier and  blockName", () => {
+    it("returns properly formatted baseclass and blockName", () => {
+        expect(bem("list-item", [], "unordered-list")).to.equal(
```

Comment:
@scooooooooby you know more about bem's intended effects than I do ... Should this be `"unordered_list unordered-list__list-item"` to match the functionality when passed a modifier?  Or should modifier-only be `list-item--scroll` ? 

Also, L13, is it missing a `list_item` or is it missing a `unordered-list`?  (or is it good the way it is?)

Reply:
Based on the order of this function I would expect line 28 to spit out `.unordered-list__list-item`, which is what @M0nica has here.

Line 13 also looks correct to me.